components/pdf_preview.jsx

```diff
@@ -149,7 +149,11 @@ export default class PDFPreview extends React.PureComponent {
             const worker = await import('pdfjs-dist/build/pdf.worker.entry.js');
             PDFJS.GlobalWorkerOptions.workerSrc = worker;
 
-            const pdf = await PDFJS.getDocument(this.props.fileUrl).promise;
+            const pdf = await PDFJS.getDocument({
+                url: this.props.fileUrl,
+                cMapUrl: '/static/cmaps/',
```

Comment:
can we import the cmaps directly from node modules here and import? Just wondering if that would work without touching the webpack config file

Reply:
Thank you. Of course I think it is better if possible, but I coudn't do it because I am not familiar with Webpack ...

Reply:
What i wanted to say that we could just import the cmaps file in `pdf_preview` and have it pass the url here. And not do anything in webpack file. Could you try that?

Reply:
Hmm...  I'm learning about Webpack (I do not have any experience of large webapps development).

My code is almost same as the instruction in [README of React-PDF](https://github.com/wojtekmaj/react-pdf/blob/main/README.md#support-for-non-latin-characters). There are 169 CMap files (with extension .bcmap) that are compressed text (postscript) files and `pdfjs-dist` loads several required CMap files from them to render a given PDF properly. As you said, if we add explicit code to import CMaps, then Webpack may automatically packs the CMap files. However, as far as I understand, since CMap files are neither .js nor .json, we must add a rule to `webpack.config.js` for using a loader or asset modules. Is it right?

Reply:
I'd have to double check this, but I think this should work how we expect based on those docs. If we were to import these, we'd have to tell Webpack how to handle them with another loader like you mentioned. That wouldn't necessarily cause them to be bundled into the JS code, but I don't know how PDFJS would expect to receive them then.

Copying them into the output folder and referring to them here should work, but we should probably make the URL here absolute since it's possible that MM isn't coming from the root of the server so this may need to be something like `https://example.com/my/mattermost/static/cmaps` in some cases. You can just use `getSiteUrl()` from `utils/url` to get that first half of the URL

Reply:
> You can just use `getSiteUrl()` from `utils/url` to get that first half of the URL

I changed as you suggested. Thank you!

Reply:
is it working with the site url direction

Reply:
At least, this code is working properly on my server `https://[mydomain]/mm`

Reply:
Awesome, thanks for double checking that case. I know that setting up MM on a subpath like that can be a bit of a pain 😄